horepg/xmltvdoc.py

```diff
@@ -1,96 +1,127 @@
 import xml.dom.minidom
 import time
-import datetime
-import calendar
+import logging
+
+
+def debug(msg):
+  logging.debug(msg)
+
+
+def warning(msg):
+  logging.warning(msg)
+
 
 class XMLTVDocument(object):
   # this renames some of the channels
   add_display_name = {}
   category_map = {
-    'tv drama': 'movie/drama (general)',
-    'actie': 'movie/drama (general)',
-    'familie': 'movie/drama (general)',
+    # 01
+    'tv drama': 'movie/drama',
+    'actie': 'movie/drama',
+    'familie': 'movie/drama',
     'thriller': 'detective/thriller',
     'detective': 'detective/thriller',
-    'avontuur': 'detective/thriller',
-    'western': 'detective/thriller',
+    'avontuur': 'adventure/western/war',
+    'western': 'adventure/western/war',
     'horror': 'science fiction/fantasy/horror',
     'sci-fi': 'science fiction/fantasy/horror',
+    'sci-fi/horror': 'science fiction/fantasy/horror',
     'komedie': 'comedy',
-    'melodrama': 'science fiction/fantasy/horror',
+    'melodrama': 'soap/melodrama/folkloric',
     'romantiek': 'romance',
     'drama': 'serious/classical/religious/historical movie/drama',
     'erotiek': 'adult movie/drama',
-    'nieuws': 'news/current affairs (general)',
+
+    # 02
+    'nieuws': 'news/current affairs',
     'weer': 'news/weather report',
     'nieuws documentaire': 'news magazine',
     'documentaire': 'documentary',
     'historisch': 'documentary',
     'waar gebeurd': 'documentary',
     'discussie': 'discussion/interview/debate',
+
+    # 03
     'show': 'game show/quiz/contest',
```

Comment:
This mapping might need to be changed to `show\game show` instead of the current mapping. For example, 'Pawn Stars' (https://en.wikipedia.org/wiki/Pawn_Stars) is being classified as a 'show' in the Horizon EPG guide, and I think the `show\game show` is more appropriate for it.

Reply:
I've struggled with this before, and concluded that mapping is sometimes not trivial. This seems to be a good approach tough.